src/index.ts

```diff
@@ -320,35 +326,30 @@ const initialize = (apiKey: string, target: Target, options?: Options): Result =
       const hasExistingFlags = !!Object.keys(evaluations).length
 
       // When authentication is done, fetch all flags
-      fetchFlags()
-        .then(() => {
-          logDebug('Fetch all flags ok', storage)
-        })
-        .then(() => {
```

Comment:
All of these promise chains by my estimation were useless and severely impacting readability. I opted to remove them and go for a much more simplified approach.